### ISSUE TITLE
DiD Sc2 - Shroud and fog is cleared when you lose

### ISSUE THREAD

**Issue by Konrad22** [wesnoth/wesnoth#3160]
Just...why? ._.

**Comment by sevu** [comment]
Isn't that always happening then you enter linger mode?

**Comment by Konrad22** [comment]
I'm not really sure what linger mode means. But I can testify now, that this only happens when you start a campaign in the multiplayer mode.

**Comment by jostephd** [comment]
@Konrad22 Linger mode is when the map is grayed out after victory or defeat but before proceeding to the next scenario (or back to the lobby).

I confirm this happens in MP but not in SP. It's due to the update_gui_to_player() call in playmp_controller::linger().

**Comment by Konrad22** [comment]
@jostephd 
Okay, thanks. ^^ So, is this intended behavior and if so, what purpose does it serve to show the full map (and the unit positions of everyone) to the player upon losing?

**Comment by jostephd** [comment]
That behavior makes sense for MP maps but I assume in SP campaigns played as MP it's unintended.

**Comment by Wedge009** [comment]
I expect the intention is to show losing players the state of the map, that they might learn who was placed where, etc.

My question is how/why are you starting a single-player game in multi-player?

**Comment by Konrad22** [comment]
Uh...Weeeeell...I wanted to show off. :c
Anyway, it's not ideal for any campaign, single or multiplayer.

**Comment by Wedge009** [comment]
I didn't realise single-player campaigns can be started from the (local) multi-player screen. Attempting to load a single-player game from the multi-player screen results in an error message. I wonder why we can now start campaigns from it.

**Comment by jostephd** [comment]
@Wedge009 The error only occurs for an SP campaign that was started from the main menu, not for an MP campaign that was started from the MP screen. It's nice to be able to play HttT S01 in coop with players controlling the elven allies :smile:

**Comment by gfgtdf** [comment]
> Anyway, it's not ideal for any campaign, single or multiplayer.

I agree here, this issue seems like a an oversight in the mpsp unififcation patch. keep in mind though that the 'start sp campaign from mp' thing is still somewhat experimental.


> I suppose there are a few times when there are allies that could be controlled by a friend in single-player campaigns.

Even if this does not casue errors, it will probably still be complteley unbalanced since the scenarios code usually assumes that the ai is stupid.

**Comment by jostephd** [comment]
When starting an SP game from the main menu, loss does _not_ reveal fog/shroud.

When starting an SP game from the MP screen, loss _does_ reveal fog/shroud.

This issue is asking to make the latter case behave like the former.

I agree with @Konrad22 that losing a scenario on a "native" MP campaigns shouldn't reveal fog/shroud either. The players would probably restart the scenario so revealing fog/shroud would be a spoiler.

**Comment by Wedge009** [comment]
Thanks for the clarification. That was my initial understanding but your 'error only occurs for an SP campaign that was started from the main menu' statement confused me as it appeared to contradict this.

**Comment by jostephd** [comment]
Oh, sorry. The error I was referring to was the one "Load Game" in the MP screen gives.
